### DISCUSSION TITLE
Very slow load of zinit.zsh

### DISCUSSION THREAD

**Discussion by dmorady1** [zdharma-continuum/zinit#426] [Q&A] [answered]
Hello, 

Is it normal that zinit.zsh is loaded that slowly and  listed on the plugin loading times?
My zshrc config for zinit is as follow, maybe I did something wrong? Any suggestion or also improvement (also generall improvements) are much appreciated!
I also tried to reinstall zinit by removing it and sourcing zshrc. No improvement was seen :cry: 
If you have an idea how to fix it, I would be happy to hear it! 
Thank you in advance!

![image](https://user-images.githubusercontent.com/29502285/202858708-cd40a13d-42ad-4c71-b58e-87f0e2852dbd.png)

``` bash
### Added by Zinit's installer
if [[ ! -f $HOME/.zinit/bin/zinit.zsh ]]; then
    print -P "%F{33}▓▒░ %F{220}Installing %F{33}DHARMA%F{220} Initiative Plugin Manager (%F{33}zdharma-continuum/zinit%F{220})…%f"
    command mkdir -p "$HOME/.zinit" && command chmod g-rwX "$HOME/.zinit"
    command git clone https://github.com/zdharma-continuum/zinit "$HOME/.zinit/bin" && \
        print -P "%F{33}▓▒░ %F{34}Installation successful.%f%b" || \
        print -P "%F{160}▓▒░ The clone has failed.%f%b"
fi

source "$HOME/.zinit/bin/zinit.zsh"
# autoload -Uz _zinit
# (( ${+_comps} )) && _comps[zinit]=_zinit

# Load a few important annexes, without Turbo
# (this is currently required for annexes)

setopt promptsubst
# first git.zsh then git
zinit wait lucid for \
                        OMZ::lib/git.zsh \
    atload"unalias grv" OMZ::plugins/git/git.plugin.zsh

zinit wait lucid for \
    OMZP::virtualenvwrapper \
    OMZP::thefuck \
    OMZP::gitignore \
    zdharma-continuum/history-search-multi-word \
    agkozak/zsh-z

zinit wait lucid light-mode for \
  atinit"zicompinit; zicdreplay" \
      zdharma-continuum/fast-syntax-highlighting \
  atload"_zsh_autosuggest_start" \
      zsh-users/zsh-autosuggestions \
  blockf atpull'zinit creinstall -q .' \
      zsh-users/zsh-completions \
      OMZP::docker/_docker

setopt auto_menu
# setopt magic_equal_subst
zmodload -i zsh/complist

# caching zsh completions
zstyle ':completion:*' accept-exact '*(N)'
zstyle ':completion:*' use-cache on
zstyle ':completion:*' cache-path ~/.zsh_cache

zstyle ':completion:*' menu select
# zstyle ':completion:*:default' menu select=2
# zstyle ':completion:*' list-colors ''

# zsh completion caseinsensitive + Would allow ex<tab> to expand to ElephantExample, but there may be undesired side effects
zstyle ':completion:*' matcher-list '' 'm:{[:lower:][:upper:]}={[:upper:][:lower:]}' \
  '+l:|?=** r:|?=**'


```

**Comment by psprint** [comment]
It's a fake plugin to be removed with:

REGISTERED_PLUGINS[-1]=()

under the `%$ZINIT[BIN_DIR]` plugin loaded at the end of zinit.zsh.

sob., 19 lis 2022, 16:39 użytkownik dmorady1 ***@***.***>
napisał:

**Reply by dmorady1** [reply]
Hello, 

thank you for your answer but it didn't help :cry: 

I added it that way (see below). But why is it there in the first place? I remember it wasn't there before and now my terminal is very slow. It takes up to 10 seconds to write anything. But it might be something else that makes my terminal slow?
Does every user have this dummy plugin?

```diff
# a searchable menu of tags for current directory
zinit null light-mode autoload'zi-browse-symbol' for %$ZINIT[BIN_DIR]
+REGISTERED_PLUGINS[-1]=()
zle -N zi-browse-symbol
zle -N zi-browse-symbol-backwards zi-browse-symbol
zle -N zi-browse-symbol-pbackwards zi-browse-symbol
zle -N zi-browse-symbol-pforwards zi-browse-symbol
zstyle -s ':zinit:browse-symbol' key ZINIT_TMP || ZINIT_TMP='\eQ'
[[ -n $ZINIT_TMP ]] && bindkey $ZINIT_TMP zi-browse-symbol

# Local Variables:
# mode: Shell-Script
# sh-indentation: 2
# indent-tabs-mode: nil
# sh-basic-offset: 2
# End:

```

**Reply by dmorady1** [reply]
Okay, I found my problem. It was a problem with an alias which was unrelated to zinit.
Does the fake plugin has an impact on my startup time?

**Comment by psprint** [comment, marked as answer]
no, it does not have any impact

On Wed, 30 Nov 2022 at 14:36, dmorady1 ***@***.***> wrote:

> Okay, I found my problem. It was a problem with an alias which was
> unrelated to zinit.
> Does the fake plugin has an impact on my startup time?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zdharma-continuum/zinit/discussions/426#discussioncomment-4274278>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABOE4CASUNDPOLO5DUYAVBLWK5Q7FANCNFSM6AAAAAASFL4KKY>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Best regards,
Sebastian Gniazdowski

**Reply by dmorady1** [reply]
Thank you very much for your help. I will mark your last comment as the answer. 
I am sorry that I did not realize that it was not a Zinit problem. But I learned that the fake plugin has no effect on the startup time. So thank you very much and have a nice day.

**Comment by vladdoster** [comment]
@dmorady1, you can reduce duplicate code with brace expansion

```zsh
zinit wait lucid for {OMZP::{'virtualenvwrapper','thefuck','gitignore'}}
```

---

Other examples I personally use in [my config](https://github.com/vladdoster/dotfiles/blob/master/zsh/.config/zsh/zinit.zsh)

## Oh-My-Zsh / Prezto

```zsh
zi is-snippet \
   light-mode \
   nocompletions \
 for {PZTM::/{'environment','history'},OMZL::{'compfix','completion','git','key-bindings'}.zsh}
```

## Annexes

```zsh
zi load for @zdharma-continuum/zinit-annex-{'patch-dl','binary-symlink','submods'}
```

## GitHub Releases 

```zsh
zi from'gh-r' \
   lbin'!' \
   nocompile \
   light-mode \
 for @{'dandavison/delta','pemistahl/grex','r-darwish/topgrade','sharkdp/'{'fd','hyperfine'}}
```

**Reply by vladdoster** [reply]
## Load time

<img width="886" alt="Screenshot 2022-12-02 at 07 14 30" src="https://user-images.githubusercontent.com/10052309/205300980-fbd0e099-fc6b-4651-a72f-d57d2b92ac9a.png">

```
ᐳ neofetch 
                    'c.          anonymous@16abc3376e91a51f 
                 ,xNMM.          -------------------------- 
               .OMMMMo           OS: macOS 13.0 22A5373b arm64 
               OMMM0,            Host: Macmini9,1 
     .;loddo:' loolloddol;.      Kernel: 22.1.0 
   cKMMMMMMMMMMNWMMMMMMMMMM0:    Uptime: 3 days, 11 hours, 33 mins 
 .KMMMMMMMMMMMMMMMMMMMMMMMWd.    Packages: 63 (brew) 
 XMMMMMMMMMMMMMMMMMMMMMMMX.      Shell: zsh 5.8.1 
;MMMMMMMMMMMMMMMMMMMMMMMM:       DE: Aqua 
:MMMMMMMMMMMMMMMMMMMMMMMM:       WM: yabai 
.MMMMMMMMMMMMMMMMMMMMMMMMX.      Terminal: /dev/ttys000 
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.    CPU: Apple M1 
 .XMMMMMMMMMMMMMMMMMMMMMMMMMMk   GPU: Apple M1 
  .XMMMMMMMMMMMMMMMMMMMMMMMMK.   Memory: 1161MiB / 8192MiB 
    kMMMMMMMMMMMMMMMMMMMMMMd
     ;KMMMMMMMWXXWMMMMMMMk.                              
       .cooc,.    .,coo:.                                
```

**Reply by dmorady1** [reply]
Thank you very much. That helps a lot!

**Reply by dmorady1** [reply]
When I'm trying to use your example. Like 
`zinit wait lucid for {OMZP::{'virtualenvwrapper','thefuck','gitignore'}}` 
I get this error. 
![image](https://user-images.githubusercontent.com/29502285/205437755-1d627537-5a2a-4b59-a245-af15294682bb.png)

Also is `zi` a shortcut for `zinit` ? Since my zshrc (vim) does not recognize `zi` .

**Reply by dmorady1** [reply]
Nevermind I found the problem with 
`zinit wait lucid for {OMZP::{'virtualenvwrapper','thefuck','gitignore'}}` 
the outer curly brackets where too much. So, it's :
`zi wait lucid for OMZP::{'virtualenvwrapper','thefuck','gitignore'}`

Now, it is much faster as before. Thank you very much ! 

![image](https://user-images.githubusercontent.com/29502285/205439532-30784c32-1de5-48ed-acd1-73adb7621628.png)
